src/common/playlist/playlist.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import {Utils} from 'playkit-js';
+import {PlaylistItem} from './playlist-item';
+
+class Playlist {
+  _id: string;
+  _items: Array<PlaylistItem>;
+  _metadata: KPPlaylistMetadata;
+  _activeItemIndex: number;
+
+  constructor() {
+    this._id = '';
+    this._items = [];
+    this._metadata = {name: '', description: ''};
+    this._activeItemIndex = -1;
+  }
+
+  configure(config: KPPlaylistConfigObject) {
+    this._id = config.id;
+    Utils.Object.mergeDeep(this._metadata, config.metadata);
+    if (config.items) {
+      this._items = [];
+      config.items.forEach(item => {
+        this._items.push(new PlaylistItem(item.sources, item.config));
```

Comment:
Why not to send the item itself in the constructor and extract the necessary data from it inside the `PlaylistItem`?